features/step_definitions/salt_steps.rb

```diff
@@ -474,11 +495,42 @@
   $server.run("salt '#{$minion_ip}' saltutil.refresh_pillar")
 end
 
-Then(/^the pillar data for "([^"]*)" should be "([^"]*)"$/) do |key, value|
-  output, _code = $server.run("salt '#{$minion_ip}' pillar.get '#{key}'")
+Then(/^the pillar data for "([^"]*)" should be "([^"]*)" on "([^"]*)"$/) do |key, value, minion|
+  if minion == "sle-minion"
+    target = $minion_ip
+    cmd = "salt"
+    extra_cmd = ""
+  elsif minion == "ceos-minion"
+    target = $ceos_minion_ip
+    cmd = "salt-ssh"
+    extra_cmd = "-i --roster-file=/tmp/tmp_roster_tests"
+    $server.run("printf '#{target}:\n  host: #{target}' > /tmp/tmp_roster_tests")
+  else
+    fail "Invalid target"
+  end
+  output, _code = $server.run("#{cmd} '#{target}' pillar.get '#{key}' #{extra_cmd}")
+  print output
   fail unless output.split("\n")[1].strip == value
 end
 
+Then(/^the pillar data for "([^"]*)" should be empty on "([^"]*)"$/) do |key, minion|
+  if minion == "sle-minion"
+    target = $minion_ip
```

Comment:
as a side remark, we have now also ssh-minion, do we test that for this target?

(centos7 minion, is a ssh or service)